File: 7. Reverse Integer/Reverse Integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        ans = 0
        num = abs(x)
        
        factor = 1 if x > 0 else -1
        
        while num != 0:
            
           
            ans = ans * 10 + (num % 10)
            
            
            num /= 10
            
        
        return 0 if ans > (pow(2, 31) - 1) else factor * ans
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        
        
        
        factor = -1 if x < 0 else 1
        
        if factor == 1:
            reverseX = int(str(x)[::-1])
        else:
            reverseX = int(str(x)[1:][::-1])
            
        
        return 0 if reverseX > (pow(2, 31) - 1) else factor * reverseX

File: 7. Reverse Integer/Reverse Integer.java
class Solution {
    public int reverse(int x) {
        int ans = 0;
        
        while(x != 0){
            if(Math.abs(ans) > Integer.MAX_VALUE / 10) return 0;
           
            ans = ans * 10 + (x % 10);
            
            x /= 10;
            
        }
        return ans;
    }
}

class Solution {
    public int reverse(int x) {
        int ans = 0;
        
        while(x != 0){
            
            int remainder = x % 10;
            int newAns = ans * 10 + remainder;
            if((newAns - remainder) / 10 != ans) return 0;
            ans = newAns;
            
            x /= 10;
            
        }
        return ans;
    }
}
